internal/services/email_change.go

```diff
@@ -0,0 +1,95 @@
+package services
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"git.epam.com/epm-lstr/epm-lstr-lc/be/pkg/domain"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+)
+
+type emailChange struct {
+	domain.Sender
+	domain.UserRepository
+	domain.EmailConfirmRepository
+	logger *zap.Logger
+	ttl    time.Duration
+}
+
+func NewEmailChangeService(senderClient domain.Sender, userRepository domain.UserRepository,
+	emailConfirmRepository domain.EmailConfirmRepository, logger *zap.Logger,
+) domain.ChangeEmailService {
+	return &emailChange{
+		Sender:                 senderClient,
+		UserRepository:         userRepository,
+		EmailConfirmRepository: emailConfirmRepository,
+		logger:                 logger,
+	}
+}
+
+func (e *emailChange) SendEmailConfirmationLink(ctx context.Context, login, email string) error {
+	e.logger.Info("new email confirmation service: send confirmation link", zap.String("login", login))
+	token := uuid.New().String()
+	user, err := e.UserRepository.UserByLogin(ctx, login)
+	if err != nil {
+		err = ErrUserNotFound
```

Comment:
Please consider returning ErrUserNotFound directly instead of the unnecessary assignment it to the err variable.